Disallow Join button without IP address input

The-Zen-Cat opened the pull request:
Closes #20234

Trying to join a server with an empty address crashes the game.  This fix disallows pressing join button without ip address field input.

PunkPun commented on OpenRA.Mods.Common/Widgets/Logic/DirectConnectLogic.cs:
```diff
@@ -39,6 +39,8 @@ public DirectConnectLogic(Widget widget, Action onExit, Action openLobby, Connec
 				portField.Text = text.Substring(last + 1);
 			}
 
+			panel.Get<ButtonWidget>("JOIN_BUTTON").IsDisabled = () => string.IsNullOrEmpty(ipField.Text);
```
It's more optimal to reuse the search

```suggestion
			var joinButton = panel.Get<ButtonWidget>("JOIN_BUTTON");
			joinButton.IsDisabled = () => string.IsNullOrEmpty(ipField.Text);
```

and reuse it on `OnClick`

PunkPun reviewed (commented):
If you write `Closes #20234` in PR description, the issue will be automatically closed on PR merge. You don't need to write on the issue itself

PunkPun commented:
Fixes #20234 also works, just without the brackets

abcdefg30 commented:
Can you squash the commits, please?

The-Zen-Cat committed:
Disallow Join button without IP address input

Disallow join button without IP address Input

Closes #20234

Trying to join a server with an empty address crashes the game. This fix disallows pressing join button without ip address field input.  Updated to reuse search for joinButton vs 2 separate calls to search.

dragunoff reviewed (approved):
LGTM :+1: 

However I think that while this works it has a problem with feedback - the user can't easily tell what is preventing them from pressing the button. It's not so evident in this example but in a more complex form where validation depends on several inputs (like the save map dialog in #20274) it may be difficult to tell where the errors are.

So we should give some feedback about what is wrong - for example in the tooltip text of the button. Perhaps an even better approach is to not disable the button, let the user press it and then highlight the invalid fields. In any case this is not something we need to solve now.

PunkPun commented:
> However I think that while this works it has a problem with feedback - the user can't easily tell what is preventing them from pressing the button. 

We could highlight the empty fields

PunkPun commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/31eec5b33bfdec67bdb5ff4d02c3bfab65358080...d21ab6b2f5ca067fcc0ae9abdf901557d51c5d3a)
